### PR TITLE
Adapt IDE to new record syntax

### PR DESCRIPTION
**Pull request by jcouv**
In particular, this removes completion for `data` and adds completion for `records`. This also fixes some other recommenders and fixers that didn't trigger on the new type syntax.

Relates to https://github.com/dotnet/roslyn/issues/40726 (test plan)

### CODE REVIEW

File: src/EditorFeatures/CSharpTest2/Recommendations/RecordKeywordRecommenderTests.cs [hunk 1]
```diff
@@ -3,13 +3,14 @@
...
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
 using Microsoft.CodeAnalysis.Test.Utilities;
 using Roslyn.Test.Utilities;
 using Xunit;
 
 namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Recommendations
 {
-    public class DataKeywordRecommenderTests : KeywordRecommenderTests
+    public class RecordKeywordRecommenderTests : KeywordRecommenderTests
```
Comment (jcouv) [inline src/EditorFeatures/CSharpTest2/Recommendations/RecordKeywordRecommenderTests.cs:13]: >RecordKeywordRecommenderTests [](start = 17, length = 29)

:memo: this file is a duplicate of ClassKeywordRecommenderTests with one or two tests added, such as typing in position nested in record.

**Comment by jcouv** [comment]
Pulled in @mavasani to help out with review since we're trying to merge this today. Thanks!

**Comment by mavasani** [comment]
@dotnet/roslyn-ide Can someone else also do a quick review as an additional sanity?

**Comment by jcouv** [comment]
@dotnet/roslyn-ide If anyone wants to take a look, please do it quick. We're aiming to get all records PRs merged by tonight.

**Pull request merged**